daemon/cmd/daemon_main.go

```diff
@@ -413,51 +410,6 @@ func InitGlobalFlags(cmd *cobra.Command, vp *viper.Viper) {
 
-	flags.Bool(option.HTTPNormalizePath, true, "Use Envoy HTTP path normalization options, which currently includes RFC 3986 path normalization, Envoy merge slashes option, and unescaping and redirecting for paths that contain escaped slashes. These are necessary to keep path based access control functional, and should not interfere with normal operation. Set this to false only with caution.")
-	option.BindEnv(vp, option.HTTPNormalizePath)
-
-	flags.String(option.HTTP403Message, "", "Message returned in proxy L7 403 body")
-	flags.MarkHidden(option.HTTP403Message)
-	option.BindEnv(vp, option.HTTP403Message)
-
-	flags.Uint(option.HTTPRequestTimeout, 60*60, "Time after which a forwarded HTTP request is considered failed unless completed (in seconds); Use 0 for unlimited")
-	option.BindEnv(vp, option.HTTPRequestTimeout)
-
-	flags.Uint(option.HTTPIdleTimeout, 0, "Time after which a non-gRPC HTTP stream is considered failed unless traffic in the stream has been processed (in seconds); defaults to 0 (unlimited)")
-	option.BindEnv(vp, option.HTTPIdleTimeout)
-
-	flags.Uint(option.HTTPMaxGRPCTimeout, 0, "Time after which a forwarded gRPC request is considered failed unless completed (in seconds). A \"grpc-timeout\" header may override this with a shorter value; defaults to 0 (unlimited)")
-	option.BindEnv(vp, option.HTTPMaxGRPCTimeout)
-
-	flags.Uint(option.HTTPRetryCount, 3, "Number of retries performed after a forwarded request attempt fails")
-	option.BindEnv(vp, option.HTTPRetryCount)
-
-	flags.Uint(option.HTTPRetryTimeout, 0, "Time after which a forwarded but uncompleted request is retried (connection failures are retried immediately); defaults to 0 (never)")
-	option.BindEnv(vp, option.HTTPRetryTimeout)
-
-	flags.Uint(option.ProxyConnectTimeout, 2, "Time after which a TCP connect attempt is considered failed unless completed (in seconds)")
-	option.BindEnv(vp, option.ProxyConnectTimeout)
-
-	flags.Uint(option.ProxyGID, 1337, "Group ID for proxy control plane sockets.")
-	option.BindEnv(vp, option.ProxyGID)
-
-	flags.Int(option.ProxyPrometheusPort, 0, "Port to serve Envoy metrics on. Default 0 (disabled).")
-	option.BindEnv(vp, option.ProxyPrometheusPort)
-
-	flags.Int(option.ProxyMaxRequestsPerConnection, 0, "Set Envoy HTTP option max_requests_per_connection. Default 0 (disable)")
-	option.BindEnv(vp, option.ProxyMaxRequestsPerConnection)
-
-	flags.Int64(option.ProxyMaxConnectionDuration, 0, "Set Envoy HTTP option max_connection_duration seconds. Default 0 (disable)")
-	option.BindEnv(vp, option.ProxyMaxConnectionDuration)
-
-	flags.Int64(option.ProxyIdleTimeout, 60, "Set Envoy upstream HTTP idle connection timeout seconds. Does not apply to connections with pending requests. Default 60s")
-	option.BindEnv(vp, option.ProxyIdleTimeout)
-
-	flags.Bool(option.DisableEnvoyVersionCheck, false, "Do not perform Envoy binary version check on startup")
-	flags.MarkHidden(option.DisableEnvoyVersionCheck)
```

Comment:
This is also a flag which was hidden and is now "un hidden". Was this intended?

Reply:
Thanks for pointing this out. But yes, this was by intention (as it might makes sense in combination with Envoy as DaemonSet). But double-checking it the wording is wrong - as it's no longer a "binary" check in all cases - will remove that word.